scipio/src/io/mod.rs

```diff
@@ -88,7 +148,45 @@ pub async fn remove<P: AsRef<Path>>(path: P) -> io::Result<()> {
     )
 }
 
-pub use self::dma_file::{Directory, DmaFile};
-pub use self::file_stream::{StreamReader, StreamReaderBuilder, StreamWriter, StreamWriterBuilder};
+pub use self::buffered_file::BufferedFile;
+pub use self::buffered_file_stream::{
+    StreamReader, StreamReaderBuilder, StreamWriter, StreamWriterBuilder,
+};
+pub use self::directory::Directory;
+pub use self::dma_file::DmaFile;
+pub use self::dma_file_stream::{
+    DmaStreamReader, DmaStreamReaderBuilder, DmaStreamWriter, DmaStreamWriterBuilder,
+};
 pub use self::read_result::ReadResult;
 pub use crate::sys::DmaBuffer;
+
+/// Allows asynchronously reading from the standard input
+pub fn stdin() -> StreamReader {
+    use std::os::unix::io::FromRawFd;
+
+    unsafe { StreamReaderBuilder::new(BufferedFile::from_raw_fd(0)).build() }
+}
```

Comment:
Hm, wouldn't `drop(stdin()); drop(stdin())` close fd 0 twice? I am also not sure how that would interract with stdlib, which also would like to own 0, 1 and 2

Reply:
You are right about dropping. Good catch.
Ultimately I don't think we should be in the business of closing 0 , 1 and 2.
I'll see if I can come up with a version of those files that don't close on drop.

As a matter of fact maybe the streams should never close, and we should push that to the `BufferedFile`. Closing is already a bit annoying right now because some of the AsyncBufExt will consume the stream, meaning it won't be able to close it... 

I'll think about it a bit more, thanks!

Note sure what stdlib does with 0 1 and 2, but if we don't close them I think this shouldn't matter. It is always legal to dispatch I/O in an open file system. The kernel doesn't really understand any of the userspace ownership...

Reply:
The cleanest way to handle this would be something akin to `mem::forget` for the file descriptor if we are using stdio fds, but `mem::forget` consumes the object so we can't keep using it. The easiest way them seem to be just special casing 0 1 and 2 inside `ScipioFile`'s `drop`

Reply:
(which can't be done, because drop has a mutable reference to self as a parameter so we can't decide not to drop the underlying file without a `mem::replace`) or similar. Rust drives me crazy sometimes.

It'd be a bit of code duplication, but the best could maybe be to implement a whole new `Stdin` type that deals directly with the `RawFd`

Reply:
This is now done